### ISSUE TITLE
Missing reviewers synchronization on server-side actions

### ISSUE THREAD

**Issue by feldgendler** [Reviewable/Reviewable#712]
When a PR is created and gets its Reviewable button, and when commits are pushed to an existing PR, the custom completion function is evaluated. However, the `pendingReviewers` it returns are not synchronized to GitHub Reviewers even if `syncRequestedReviewers` is set.

This seems to cause long-term mismatches between `pendingReviewers` and GitHub reviewers.

**Comment by pkaminski** [comment]
I pushed a build that will attempt to respect `syncRequestedReviewers` on the server.  This only works if the repo is connected and the request will appear to come from one of the repo admins (usually the one who connected the repo, but not necessarily).  Note that this sync is not necessarily retroactive:  old PRs will not be automatically updated, but if something else triggers an update then reviewers will be requested.  New PRs should be fine.

Let me know how it goes!  I may need to temporarily roll back if I spot anything weird, though.

**Comment by feldgendler** [comment]
When `syncRequestedReviewers` isn't forced one or another way by the custom function, the default setting of the sync checkbox under “Publish” is “checked”. But to see the effect of this fix, one has to force `syncRequestedReviewers: true` by the custom function, meaning that the default is _not_ to sync.

This is inconsistent. The default for client-side and server-side operations should be the same.

**Comment by pkaminski** [comment]
Yeah, I agree this is awkward.  However, while I'm OK defaulting an easily visible checkbox to true, I don't want to default a very hidden setting to true (especially when this has historically been false).  I think the right solution is to replace both the per-publish checkbox and the completion flag with a visible repo-level setting that takes the place of both (and removes control from individual reviewers).

I'll reopen this issue to track this but don't know when I'll have time to implement it.

**Comment by feldgendler** [comment]
Yes, please. I don't see a reason to control this per occasion, either.
